packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -931,6 +941,49 @@ class _CupertinoDatePickerDateState extends State<CupertinoDatePicker> {
 // layout will be broken.
 
 
+/// Controls a [CupertinoTimerPicker] widget.
+///
+/// Each controller can only be used with a single Cupertino timer picker.
+///
+/// Used with [CupertinoTimerPicker].
+///
+/// See also:
+///
+///  * [CupertinoTimerPicker] which creates a Cupertino themed time picker which
+///    uses this controller.
+///  * [CupertinoDatePicker] which creates a Cupertino themed date picker.
+///
+class CupertinoTimerPickerController extends ChangeNotifier {
+  /// Creates a CupertinoTimerPickerController.
+  ///
+  /// The [resetAnimationDuration] and [desiredTimerDuration] arguments must not be null.
+  CupertinoTimerPickerController({
+    Duration desiredTimerDuration = Duration.zero,
+    this.resetAnimationDuration = const Duration(milliseconds: 300),
+  }) : assert(resetAnimationDuration != null),
+       assert(desiredTimerDuration != null) {
+    _desiredTimerDuration = desiredTimerDuration;
```

Comment:
In general, assignments like these can be in the initializer (along with the asserts) rather than be in the constructor.

Reply:
Ok, so when do we want to do things in the initializer vs the constructor? This is how it was done in editable.dart.